Sql statement api error messaging fixes. (#14629)
apache/druid: docs/multi-stage-query/reference.md

For queries where you want to use fault tolerance for workers,  set `faultTolerance` to `true`, which automatically sets `durableShuffleStorage` to `true`.

> For select queryies which want to write the final result to `durableStoage`, set `selectDestination`:`durableStorage`. Which shuffle mesh the job uses can still be controller by `durableShuffleStorage` flag ie. a combination where `selectDestination`:`durableStorage` and `durableShuffleStorage`:`false` is perfectly valid.

Reviewer: 1. spelling of queries and durableStorage
2. Something seems off in the second sentence. I think it should be "controlled" instead of "controller".
3. `durableStoage` (first use) should be written as "durable storage" (without double quotes) since we are using it as a noun instead of code word.
4. We should use something else instead of "shuffle mesh" in the docs since it's not a well-documented term in the docs and could be simplified further. Wdyt about simplifying it to something with following structure:

```
Set `selectDestination`:`durableStorage` for select queries that want to write the final
results to durable storage instead of the task reports. Saving the results in the durable
storage allows users to .. (benefits for the user)

The location where the workers can write the intermediate results is independent of the
location where the final results get stored. Therefore "durableShuffleStorage":false and
"selectDestination":"durableStorage" is a valid configuration to use in the query context, that
instructs the controller to persist only the final result in the durable storage, and not the
intermediate results. 
```

Author: I like you structure better. Going ahead with that.

For queries where you want to use fault tolerance for workers,  set `faultTolerance` to `true`, which automatically sets `durableShuffleStorage` to `true`.

Set `selectDestination`:`durableStorage` for select queries that want to write the final results to durable storage instead of the task reports. Saving the results in the durable

> storage allows users to fetch large result sets. The location where the workers write the intermediate results is different than the location where final results get stored. Therefore, `durableShuffleStorage`:`false` and

Reviewer: ```suggestion
storage allows users to fetch large result sets. The location where the workers write the intermediate results can be different from the location where the final results get stored. Therefore, `durableShuffleStorage`:`false` and
```

Author: The location where the workers write the intermediate results is always different hence I would like to keep the wording as is unless you feel strongly about it.
